interfaces/cython/cantera/base.pyx

```diff
@@ -349,8 +379,20 @@ cdef class _SolutionBase:
             for i,spec in enumerate(species):
                 self._selected_species[i] = self.species_index(spec)
 
-    def __reduce__(self):
-        raise NotImplementedError('Solution object is not picklable')
+    def __getstate__(self):
+        """Save complete information of the current phase for pickling."""
+        if self.kinetics.nTotalSpecies() > self.thermo.nSpecies():
+            raise NotImplementedError(
+                "Pickling of Interface objects is not implemented.")
+        transport_model = pystr(self.transport.transportType())
+        return self.write_yaml(), self.state, transport_model
```

Comment:
The YAML definition already includes the state information, and also the transport model. Is there a reason to duplicate that in the pickle? Thanks to some other changes in this PR, the following works and matches both properties:
```python
gas2 = ct.Solution(yaml=gas.write_yaml())
```

Reply:
Ha. good catch!